packages/core/src/browser/authentication-service.ts

```diff
@@ -99,72 +130,84 @@ export interface AuthenticationService {
-                    const menuAction = this.menus.registerMenuAction(subSubMenuPath, {
-                        label: 'Sign Out',
-                        commandId
-                    });
-                    disposables.push(menuAction);
-                    disposables.push(command);
-                    disposableMap.set(session.id, disposables);
-                });
-            }
-            if (e.event.removed.length > 0) {
-                e.event.removed.forEach(removed => {
-                    const toDispose = disposableMap.get(removed);
-                    if (toDispose) {
-                        toDispose.dispose();
-                        disposableMap.delete(removed);
-                    }
-                });
-            }
-        });
+        this.onDidChangeSessions(event => this.handleSessionChange(event));
         this.commands.registerCommand(this.noAccountsCommand, {
             execute: () => { },
             isEnabled: () => false
         });
     }
 
+    protected async handleSessionChange(changeEvent: SessionChangeEvent): Promise<void> {
+        if (changeEvent.event.added.length > 0) {
+            const sessions = await this.getSessions(changeEvent.providerId);
+            sessions.forEach(session => {
+                if (!this.sessionMap.get(session.id)) {
+                    this.sessionMap.set(session.id, this.createAccountUi(changeEvent.providerId, changeEvent.label, session));
+                }
+            });
+        }
+        for (const removed of changeEvent.event.removed) {
+            const sessionId = typeof removed === 'string' ? removed : removed?.id;
+            if (sessionId) {
+                this.sessionMap.get(sessionId)?.dispose();
+                this.sessionMap.delete(sessionId);
+            }
+        }
+    }
+
+    protected createAccountUi(providerId: string, providerLabel: string, session: AuthenticationSession): DisposableCollection {
+        // unregister old commands and menus if present (there is only one per account but there may be several sessions per account)
+        const providerAccountId = `account-sign-out-${providerId}-${session.account.id}`;
+        this.commands.unregisterCommand(providerAccountId);
+
+        const providerAccountSubmenu = [...ACCOUNTS_SUBMENU, providerAccountId];
+        this.menus.unregisterMenuAction({ commandId: providerAccountId }, providerAccountSubmenu);
+
+        // register new command and menu entry for the sessions account
+        const disposables = new DisposableCollection();
+        disposables.push(this.commands.registerCommand({ id: providerAccountId }, {
+            execute: async () => {
+                this.signOutOfAccount(providerId, session.account.label);
+            }
+        }));
+        this.menus.registerSubmenu(providerAccountSubmenu, `${session.account.label} (${providerLabel})`);
+        disposables.push(this.menus.registerMenuAction(providerAccountSubmenu, {
+            label: 'Sign Out',
```

Comment:
One last thing, this should be localized:
```suggestion
            label: nls.localizeByDefault('Sign Out'),
```

Reply:
Done! I actually found a few more places I internationalized. Hopefully it is fine now.